### DISCUSSION TITLE
Branch policy

### DISCUSSION THREAD

**Discussion by yangmr04** [wexlergroup/FreeBird.jl#2] [Announcements]
![Git Flow](https://numergent.com/images/dev/git-flow.png)

1. The `main` branch is for production code only. 
2. The `dev` branch is for active code development and features accumulation. Once a major change has happened, or several new features are collected, merge it into `main` via a Pull Request, and tag a new version.
3. The `feature/*` branches are for implementations of new untested features. When the feature is ready, merge it into `dev` via a Pull Request.
4. The `hotfix/*` branches are for hot-fixes and should merge into `dev` via a Pull Request.
5. The `docs/*` branches are for documentation-related changes only. Use them if you change anything in the `docs` directory, and the docstrings of functions (only the docstrings, not the function definitions!). 

The logic is based on [A successful Git branching model](https://nvie.com/posts/a-successful-git-branching-model/). Please follow these policies and naming conventions to keep the repo clean and readable for everyone, both users and developers.

**Comment by rwexler** [comment]
Hi @yangmr04, thanks for sharing this! Can you define a hotfix?

**Reply by yangmr04** [reply]
A hotfix is a change (or changes) to the code outside the normal development flow with urgency that addresses bugs in the code. Let's say if we discover that the Lennard-Jones potential is not defined correctly, where the first term is raised to the power of 13 instead of 12, we should open a `hotfix/fix-lj-potential-power` branch to simply correct the term, and the change should be merged directly to `main` (and `dev`) without going though typical feature -> develop -> release -> production cycle. This way the users can get the fix as soon as possible, because longer the incorrect implementation floats around, more damages it will cause; Thus, a "hot"-fix is needed.
